test/parallel/test-http-client-timeout-on-connect.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  // This space is intentionally left blank.
+});
+
+server.listen(0, common.localhostIPv4, common.mustCall(() => {
+  const port = server.address().port;
+  const req = http.get(`http://${common.localhostIPv4}:${port}`);
+
+  req.setTimeout(1);
+  req.on('socket', common.mustCall((socket) => {
+    assert.strictEqual(socket._idleTimeout, undefined);
+    socket.on('connect', common.mustCall(() => {
+      assert.strictEqual(socket._idleTimeout, 1);
+    }));
+  }));
+  req.on('timeout', common.mustCall(() => req.abort()));
+  req.on('error', common.mustCall((err) => {
+    assert.ok(err);
```

Comment:
It doesn't hurt anything, but you don't really need this assertion since the next one would handle this.